scripts/tsv_to_mrmatrix.py

```diff
@@ -42,60 +42,37 @@ def coarsen(f):
         dask_dset = da.coarsen(np.nansum, dask_dset, {0: 2, 1: 2})
         da.store(dask_dset, values)
 
-def main():
-    parser = argparse.ArgumentParser(description="""
-    
-    python tsv-dense-to-sparse
-""")
 
-    parser.add_argument('input_file')
-    parser.add_argument('output_file')
-    #parser.add_argument('-o', '--options', default='yo',
-    #					 help="Some option", type='str')
-    #parser.add_argument('-u', '--useless', action='store_true',
-    #					 help='Another useless option')
-    parser.add_argument('-n', '--first-n', type=int, default=None,
-            help="Only use the first n entries in the matrix")
-
-    args = parser.parse_args()
-
-    count = 0
-    top_n = args.first_n
-
-    if args.input_file == '-':
-        f_in = sys.stdin
-    else:
-        f_in = open(args.input_file, 'r')
-
-    first_line = next(f_in)
-    parts = first_line.split('\t')
+def parse(input_handle, output_hdf5, top_n=None):
+    input_handle
+    first_line = next(input_handle)
+    parts = first_line.strip().split('\t')
+    # TODO: Use the python built-in csv module, instead of parsing by hand?
 
     if top_n is None:
         top_n = len(parts) - 1
+        # TODO: So if it's taller than it is wide, it will be truncated to a square,
+        # unless an explicit top_n is provided? That doesn't seem right.
```

Comment:
You're right. I wrote this script with square matrices in mind. Extending it to rectangular matrices will need some fixing. Having a TODO is a nice way of kicking that can down the road.